docs/src/rules/prefer-object-spread.md

```diff
@@ -55,6 +55,9 @@ Object.assign(foo, bar);
 Object.assign(foo, { bar, baz });
 
 Object.assign(foo, { ...baz });
+
+var foo = Object.assign;
+var bar = foo({}, baz);
```

Comment:
I feel this should be reported because foo has never been reassigned.

**Incorrect**

```js
/*eslint prefer-object-spread: "error" */

let doSomething;

doSomething = Object.assign;

var bar = doSomething({}, a, b); // Error
```
**Correct**
```js
/*eslint prefer-object-spread: "error" */

let doSomething;

if (foo) {

doSomething = Object.assign;
} else {
doSomething = somethingElse
}
var bar = doSomething({}, a, b); // OK
```

Reply:
seems like i have understood the issue wrong but what about autofixes is it also correct
```js
/*eslint prefer-object-spread: "error" */

let doSomething;

doSomething = Object.assign;

var bar = { ...a, ...b}; 
```

Reply:
Yes, I believe the auto fix is also correct, `doSomething` will be reported by the `no-unused-vars` rule. 

@mdjermanovic Can you once confirm the expected behavior?

Reply:
The problem is that `ReferenceTracker` returns both certain references (e.g., `const x = Object.assign; x();`) and potential references (those that are references only in some code paths, e.g., `const x = foo ? Object.assign : bar; x();`), which is okay for some rules (e.g., for `no-obj-calls`, because it's a bug if  `JSON()` is called in any code paths) but not for this one.

We currently don't have a utility that would help find only certain references. The solution could be:

1. Add an option to `ReferenceTracker` to not return potential references.
2. Make another utility.
3. Limit this rule to check only explicit `Object.assign` calls. Maybe also `<global object>.Object.assign` (like `no-eval`).

Reply:
So, changes in this PR falls in 3rd point so what should we do now, move forward or look for another approaches (point 1st and 2nd).
the reason i chose 3rd one is i noticed in this rule's previous version is that references was used so that it allows the `Object.assign` if `Object` is importing from a module, and i didn't find any tests which doesn't call `Object.assign` directly.

Reply:
I am still doubtful that this fix is correct and doesn't lead to bugs.
```js
/*eslint prefer-object-spread: "error" */

let doSomething;

doSomething = Object.assign;

var bar = { ...a, ...b}; 
```

Reply:
Perhaps we could first ask at https://github.com/eslint-community/eslint-utils if option 1 would be possible. If not, then I think option 3 is fine.